### PR TITLE
Fix cmip6 postprocessing

### PR DESCRIPTION
**Pull request by TonyB9000**
Please review, post questions, etc.

### CODE REVIEW

File: warehouse/warehouse/resources/dataset_spec.yaml
Comment (chengzhuzhang) [inline warehouse/warehouse/resources/dataset_spec.yaml:1590]: `piControl-ext85yr` is not one of esgf facet names, not sure how this works.I think the extension was appended to the original piControl run when publishing?

File: warehouse/warehouse/resources/dataset_spec.yaml
Comment (chengzhuzhang) [inline warehouse/warehouse/resources/dataset_spec.yaml:64]: Removing _highfreq will only work with e2c code on master once Tom's PR is merged.

File: warehouse/warehouse/tools/list_cmip6_dsids.py
Comment (chengzhuzhang) [inline warehouse/warehouse/tools/list_cmip6_dsids.py:3]: It would be helpful to add some comments on how to use this script.

**Review by chengzhuzhang — COMMENTED** [review]
Looking good. Just some minor comments! Thank you, Tony.

File: warehouse/warehouse/resources/dataset_spec.yaml
Comment (TonyB9000) [reply warehouse/warehouse/resources/dataset_spec.yaml:64]: I eliminated the "_highfreq" in the dataset_spec  many weeks ago (so that dataset_ids could be cleanly generated) and then adjusted for that in GenerateAtmDay, GenerateAtm3hr, with

>             vname = item['CMIP6 Name']
>             if vname in ['pr', 'rlut']:
>                 vname = f"{vname}_highfreq" # only for this day or 3hr job

So that e2c would be happy.  I will need to eliminate that once e2c takes care if it internally.

Comment (TonyB9000) [reply warehouse/warehouse/resources/dataset_spec.yaml:64]: P.S.  That "item['CMIP6 Name']" comes from "e3sm_to_cmip --info".

File: warehouse/warehouse/tools/list_e3sm_dsids.py
Comment (TonyB9000) [reply warehouse/warehouse/tools/list_e3sm_dsids.py:3]: I will add a "-h".  Presently, they are hard-coded for simple "python list_e3sm_dsids.py" usage.  There is no "main", There are no options or command line parameters.

Comment (TonyB9000) [reply warehouse/warehouse/tools/list_e3sm_dsids.py:21]: Done.  Both now provide -h/--help usage.

Comment (chengzhuzhang) [reply warehouse/warehouse/tools/list_e3sm_dsids.py:21]: Got it. I guess I saw some example usage you explained to me as follows:
For E3SM, we could isolate all such dataset_ids (only “time-series” are per-variable) by the command
 
python list_e3sm_dsids.py | grep time-series | fgrep ".land." > land_var_dsids_E3SM
 
For CMIP6 sets:
 
python list_cmip6_dsids.py | fgrep “.Lmon.” > land_var_dsids_CMIP6

Good to have this either in the command line or in help message. But I think it is also fine without it.

**Review by chengzhuzhang — APPROVED** [review]

File: warehouse/warehouse/tools/list_e3sm_dsids.py
Comment (TonyB9000) [reply warehouse/warehouse/tools/list_e3sm_dsids.py:21]: Good point.  I could easily expand the help-text.  Also (eventually) add fancy filters to the command line, like

--select realm=land
--omit freq=day,3hr

etc.

File: warehouse/warehouse/resources/dataset_spec.yaml
Comment (chengzhuzhang) [reply warehouse/warehouse/resources/dataset_spec.yaml:1590]: @TonyB9000 would you clarify on this point?

Comment (TonyB9000) [reply warehouse/warehouse/resources/dataset_spec.yaml:1590]: I'm sure that was a copy/paste error.  (for cmip-case).  The "cmip-case" for both "piControl" and "piControl-ext85yr" could be the same.

Perhaps we can eliminate the entire "case"?  But if we are to publish any of the "ex85yr" sets separately of the original, they need their own dataset_ids, and need to be represented separately in the dataset_spec.

Do I understand your question correctly?

Comment (chengzhuzhang) [reply warehouse/warehouse/resources/dataset_spec.yaml]: So -ext85yr in both cmip_case (CMIP6.CMIP.E3SM-Project.E3SM-1-1.piControl and CMIP6.CMIP.E3SM-Project.E3SM-1-1-ECA.piControl) should be removed? or they are fine as is. If i remembered correctly i think those ext85yr runs were stitched to the original publication, and republished.

Comment (TonyB9000) [reply warehouse/warehouse/resources/dataset_spec.yaml]: I have reverted that cmip_case to CMIP6.CMIP.E3SM-Project.E3SM-1-1.piControl

Comment (TonyB9000) [reply warehouse/warehouse/resources/dataset_spec.yaml]: No - that was the original "quest", but the myriad difficulties that arose when trying to stitch together runs made in differnt years, on different machines, by different people, led to too many obstacles.  So, we (including Sterling) made the decision to publish ALL "extension" sets as separate datasets.  Hence, they need separate dataset_ids.  Therefore, the (project: CMIP6) portion of the dataset_spec reads, in part:

            E3SM-1-1:
                piControl:
                    start: 1850
                    end: 2014
                    ens:
                        - r1i1p1f1
                    except:
                        - siu
                        - siv
                        - clcalipso
                **piControl-ext85yr**:
                    start: 2015
                    end: 2100
                    ens:
                        - r1i1p1f1
                    except:
                        - siu
                        - siv
                        - clcalipso
                historical:
                    start: 1850
                    end: 2014
                    ens:
                        - r1i1p1f1
                    except:
                        - siu
                        - siv
                        - clcalipso
            E3SM-1-1-ECA:
                piControl:
                    start: 1850
                    end: 2014
                    ens:
                        - r1i1p1f1
                    except:
                        - siu
                        - siv
                        - clcalipso
                **piControl-ext85yr**:
                    start: 2015
                    end: 2100
                    ens:
                        - r1i1p1f1
                    except:
                        - siu
                        - siv
                        - clcalipso
                historical:
                    start: 1850
                    end: 2014
                    ens:
                        - r1i1p1f1
                    except:
                        - siu
                        - siv
                        - clcalipso

Comment (chengzhuzhang) [reply warehouse/warehouse/resources/dataset_spec.yaml]: Thanks, and i thinks there is another instance for CMIP6.CMIP.E3SM-Project.E3SM-1-1-ECA.piControl https://github.com/E3SM-Project/esgfpub/blob/00586cb6ffb38afe6998b62727eb0d814d6c4b9b/warehouse/warehouse/resources/dataset_spec.yaml#L2087

Comment (TonyB9000) [reply warehouse/warehouse/resources/dataset_spec.yaml]: Now I think I need to switch the cmip_case back, or else the "years" will not align.

Comment (TonyB9000) [reply warehouse/warehouse/resources/dataset_spec.yaml]: Note:  Technically, once could publish the raw E3SM datasets separately, but publish the CMIP6 sets as a single "1850-2100" body.  But the DataSM will not support that - at present it does not even support generating "pbo/pso" due to the way it attempts to locate the proper (ocean + atmos) datasets needed.  I am struggling with that right now (for the past 48 hours.)

Comment (TonyB9000) [reply warehouse/warehouse/resources/dataset_spec.yaml]: Yes, one for "1_1" (CTC) and the other for "1_1_ECA".  The "yaml" format makes that VERY hard to check.  I think using the "list_e3sm_dsids | cut -f1-3 -d. | sort | uniq | grep ext85yr" is the best way to see it.  That yields:

E3SM.1_1_ECA.piControl-ext85yr
E3SM.1_1.piControl-ext85yr

I wish I had the time to produce a flatter structure that I could both read and write, and convert between.  At present, too many codes depend upon the yaml format to dispense with it.  But editing that is a real pain.  Even with those "list_dsid" apps, you cannot see the "cmip_case" values, so you cannot easily tell if each is properly associated with each dataset.

**Comment by TonyB9000** [comment]
@tomvothecoder Hi Tom - should I merge or wait?  Thanks!

**Review by tomvothecoder — APPROVED** [review]

**Comment by tomvothecoder** [comment]
Hi @TonyB9000, if you don't have anymore changes, this looks ready to merge.

Afterwards, I will rebase PR #107 on top of it and proceed.

**Comment by TonyB9000** [comment]
OK, once I merge, should I wipe this branch, and pull master?  I don't do this often enough to get it straight...

Also, should I hold off on crafting yet-another-branch until you are done with any translations?  I'd feel most comfortable with a fresh git-clone.

**Comment by TonyB9000** [comment]
When it says "the ... branch can be safely deleted", this is the remote speaking.  The local should be deleted as well, right?

**Comment by tomvothecoder** [comment]
> OK, once I merge, should I wipe this branch, and pull master? I don't do this often enough to get it straight...

You are correct. I would also run `git fetch --prune` to removed deleted remote branches from your local git's list of branches.

> Also, should I hold off on crafting yet-another-branch until you are done with any translations? I'd feel most comfortable with a fresh git-clone.

I think that is a good idea because rebasing PR #107 with any changes to files under the old `warehouse` directory isn't the most straightforward to sort out, since the directory is now being called `datasm`.

> When it says "the ... branch can be safely deleted", this is the remote speaking. The local should be deleted as well, right?

Yes to both. If you ever need to restore the PR's branch, it can be done through the PR.

**Pull request merged**
